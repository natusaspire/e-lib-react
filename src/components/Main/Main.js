import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Home from '@/components/Home';
import LibraryContainer from '@/containers/Library';
import MaterialContainer from '@/containers/Material';
import Error from '@/components/Error';

const renderError = () => <Error status={404} />;

const Main = () => (
  <Switch>
    <Route
      exact
      path="/"
      component={Home}
    />
    <Route
      exact
      path="/library"
      component={LibraryContainer}
    />
    <Route
      exact
      path="/materials/:endpoint/:id"
      component={MaterialContainer}
    />
    <Route
      path="**"
      render={renderError}
    />
  </Switch>
);

export default Main;
