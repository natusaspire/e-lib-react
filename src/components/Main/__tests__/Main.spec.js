import React from 'react';
import { shallow } from 'enzyme';

import Main from '@/components/Main';

jest.mock('react-router-dom', () => ({
  Switch: 'Switch',
  Route: 'Route'
}));

jest.mock('@/components/Home', () => 'Home');
jest.mock('@/containers/Library', () => 'LibraryContainer');
jest.mock('@/containers/Material', () => 'MaterialContainer');
jest.mock('@/components/Error', () => 'Error');

describe('<Main />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Main />);

    expect(wrapper).toMatchSnapshot();
  });
});
