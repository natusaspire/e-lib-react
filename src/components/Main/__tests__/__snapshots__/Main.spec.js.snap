// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render main content 1`] = `
<Switch>
  <Route
    component="Home"
    exact={true}
    path="/"
  />
  <Route
    component="LibraryContainer"
    exact={true}
    path="/library"
  />
  <Route
    component="MaterialContainer"
    exact={true}
    path="/materials/:endpoint/:id"
  />
  <Route
    path="**"
    render={[Function]}
  />
</Switch>
`;
